internal/pkg/cli/task_run.go

```diff
@@ -302,6 +308,39 @@ func (o *runTaskOpts) Validate() error {
 		}
 	}
 
+	if noOS, noArch := o.os == "", o.arch == ""; noOS != noArch {
+		return fmt.Errorf("must specify either both `--%s` and `--%s` or neither", osFlag, archFlag)
+	}
+	if err := o.validateOS(); err != nil {
+		return err
+	}
+	if err := o.validateArch(); err != nil {
+		return err
+	}
+
+	// Set default values for CPU and memory based on the operating system.
+	// The flag default value is 0. If a user sets `--cpu 0` or `--memory 0`, then the 0 value will be replaced by 256/1024 for cpu and 512/2048 for mem, depending on OS, instead of erroring out.
+	if o.cpu == 0 {
+		o.cpu = 256
+		if isWindowsOS(o.os) {
+			o.cpu = manifest.WindowsTaskCPU
+		}
```

Comment:
This looks a bit off, since we are not supposed to assign values in `Validate()`...

Reply:
Agreed! I think we should keep the default cpu and mem settings as is:
https://github.com/aws/copilot-cli/blob/d5c08711e55f9b346ecb49180c494b4e027f51da/internal/pkg/cli/task_run.go#L865-L866

It's a bit odd to read the default as 0 as user, they might think it's a legit accepted value. Instead, what do you think of returning an error here if `cpu` is not at least `MinWindowsTaskCPU`?

Reply:
Also the user might be surprised to see a higher bill if we change the cpu and mem setting without asking them first.